src/Compilers/CSharp/Portable/Symbols/BaseTypeAnalysis.cs

```diff
@@ -35,12 +35,22 @@ private static void ClassDependsClosure(NamedTypeSymbol type, CSharpCompilation
             type = type.OriginalDefinition;
             if (partialClosure.Add(type))
             {
-                ClassDependsClosure(type.GetDeclaredBaseType(null), currentCompilation, partialClosure);
+                if (type.IsInterface)
+                {
+                    foreach (var bt in type.GetDeclaredInterfaces(null))
+                    {
+                        TypeDependsClosure(bt, currentCompilation, partialClosure);
+                    }
+                }
+                else
+                {
+                    TypeDependsClosure(type.GetDeclaredBaseType(null), currentCompilation, partialClosure);
+                }
 
                 // containment is interesting only for the current compilation
                 if (currentCompilation != null && type.IsFromCompilation(currentCompilation))
                 {
```

Comment:
I don't see why this condition should be present.  There may be a compat issue for classes, but why is it here for interfaces?

Reply:
Oh, it is because nothing in that other assembly can depend on the current compilation in any way.